### ISSUE TITLE
Rename Widget Types?

### ISSUE THREAD

**Issue by zicklag** [PsichiX/raui#10]
Currently we have different concepts that are all called Widgets:

- **Widget Nodes**
- **Widget Components**
- **Widget Units**

This becomes a little confusing because the term "Widgets" could refer to any of them. I was wondering whether or not it made sense to rename them to make the concepts more clear:

- **Widget Nodes** ⇒ **Tree** ( the type could still be `WidgetTree` because it is a tree of `Widget`s )
- **Widget Component** ⇒ **Widget** ( or _possibly_ **Component** )
- **Widget Unit** ⇒ **Atom**

I think this would help make things a little less confusing.

For instance, right now we have the `widget!` macro which returns a `WidgetNode` or sometimes a `WidgetUnit`, but we also have the `make_widget` macro, which returns a `WidgetComponent`. It's all just a little confusing and I got stuck for just a second trying to figure out the difference between a `WidgetNode` and a `WidgetComponent` and why a widget component returns a `WidgetNode`.

We could change this so that we have a `tree!` macro which returns a `WidgetTree`, the `widget!` macro ( which is currently called `make_widget` ) that returns a `Widget`, and an `atom!` macro that returns an `Atom`. I think this would help solidify the concepts a little bit more with less confusion.

---

:man_facepalming: I just realized a `WidgetNode` isn't a tree. That's how useful writing documentation is to help you understand things. I realized it while writing the note on "what is a widget node?" LOL.

OK, maybe something like this:

- **Widget Node** ⇒ **Widget**
- **Widget Component** ⇒ **Component**
- **Widget Unit** ⇒ **Atom**

Then we can explain it like:

- **Widget:** A `Widget` represents any one of the three different kinds of widgets: `None`, `Atom`, or `Component`.
- **Atom:** An `Atom` is a primitive widget such as a `TextBox`,  `ImageBox`, or `FlexBox`. There are small number of different kinds of `Atom`s and the RAUI rendering backends are repsonsible for being able to render the atoms.
- **Component:** A `Component` is a widget that is made out of a function that takes inputs such as properties and the slots for its child widgets and that returns an output widget as a result. Components can return atoms or other components and when rendering RAUI will convert the component tree into a tree of `Atom`s for the rendering backend to display.

**Comment by PsichiX** [comment]
i'm ok with changing the names, but what about names of the types itself? when ppl will use RAUI along with some ECS crates where Component is also a thing they might get confused, how we are gonna fit type names into that change and not confuse users as we just moved from macro based components into function based components?

**Comment by zicklag** [comment]
The point about `Component` is a good one. :thinking:

You know what, after I got a better understanding of what these actually were, the names _do_ make sense. And the explanations you gave for them in the README were even  good, I just got a little confused specifically with WidgetNode once I got into the code.

At this point I'm thinking just leave the names for now. Having the Getitng Started guide that kind of walks through actual usage piece by piece will probably help with any possible confusion.
